packages/ui/react-ui-editor/src/components/TextEditor/extensions/cursor-converter.ts

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { Facet } from '@codemirror/state';
+
+/**
+ * Converts indexes into the text document into stable peer-independent cursors.
+ *
+ * See:
+ *  - https://automerge.org/automerge/api-docs/js/functions/next.getCursor.html
+ *  - https://github.com/yjs/yjs?tab=readme-ov-file#relative-positions
+ *
+ * @param {assoc} number Negative values will associate the cursor with the previous character
+ *                       while positive - with the next one.
+ */
+export interface CursorConverter {
```

Comment:
move file into util.ts (would like to keep top-level files as extensions -- otherwise require they all have folders).